x-pack/plugins/discover_log_explorer/public/customizations/custom_dataset_selector.tsx

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { DiscoverStateContainer } from '@kbn/discover-plugin/public';
+import { DatasetSelectionHandler, DatasetSelector } from '../components/dataset_selector';
+import { DatasetsProvider, useDatasetsContext } from '../hooks/use_datasets';
+import { InternalStateProvider, useDataView } from '../hooks/use_data_view';
+import { IntegrationsProvider, useIntegrationsContext } from '../hooks/use_integrations';
+import { IDatasetsClient } from '../services/datasets';
+
+interface CustomDatasetSelectorProps {
+  stateContainer: DiscoverStateContainer;
+}
+
+export const CustomDatasetSelector = withProviders(({ stateContainer }) => {
+  // Container component, here goes all the state management and custom logic usage to keep the DatasetSelector presentational.
+  const dataView = useDataView();
+
+  const {
+    error: integrationsError,
+    integrations,
+    isLoading: isLoadingIntegrations,
+    loadMore,
+    reloadIntegrations,
+    searchIntegrations,
+    searchIntegrationsStreams,
+    sortIntegrations,
+    sortIntegrationsStreams,
+  } = useIntegrationsContext();
+
+  const {
+    datasets,
+    error: datasetsError,
+    isLoading: isLoadingStreams,
+    loadDatasets,
+    reloadDatasets,
+    searchDatasets,
+    sortDatasets,
+  } = useDatasetsContext();
+
+  /**
+   * TODO: this action will be abstracted into a method of a class adapter in a follow-up PR
+   * since we'll need to handle more actions from the stateContainer
+   */
+  const handleStreamSelection: DatasetSelectionHandler = (dataset) => {
+    return stateContainer.actions.onCreateDefaultAdHocDataView({
```

Comment:
I think we discussed the option of encoding the integration name and dataset in the data view id so we can reconstruct it on reload. Do you think that should be tackled in a follow up or as part of this PR?

Reply:
I think it could be already part of this work since it should be a big effort.
In terms of how we generate the id, what about using the integrationId format (`integration-<integration-name>-<integration-version>`) and concatenate the dataset name? Or maybe is better to use the index pattern?

Something like `integration-system-1.33.0-application`? 
Also, shall we leave it as a raw string or encode it somehow (base64, etc..)?

Reply:
Based on the external talks we had about the topic, we may want to talk more about this before implementing a solution, I'll suggest keeping this as part of a follow-up PR, wdyt?

Reply:
Sounds good to me. IMHO we should prioritize it highly, though.